tests/test_modeling_common.py

```diff
@@ -1215,15 +1215,39 @@ def _create_and_check_torch_fx_tracing(self, config, inputs_dict, output_loss=Fa
                             (past_mask, inputs_to_test[1]["attention_mask"]), dim=1
                         )
 
-            if "inputs_embeds" in inspect.signature(model.forward).parameters and not model.config.is_encoder_decoder:
-                inputs_to_test.append(
-                    {
-                        "inputs_embeds": torch.rand(
-                            2, 2, model.config.hidden_size, dtype=torch.float, device=torch_device
-                        )
-                    }
+            if (
+                "input_ids" in inspect.signature(model.forward).parameters
+                and "inputs_embeds" in inspect.signature(model.forward).parameters
+                and not model.config.is_encoder_decoder
+            ):
+                inps = copy.deepcopy(inputs_to_test[0])
+
+                embedding_size = (
+                    model.config.embedding_size
+                    if getattr(model.config, "embedding_size", None) is not None
+                    and model.config.model_type != "megatron-bert"
+                    else model.config.hidden_size
                 )
 
+                if (
+                    model.config.model_type in MODEL_FOR_MULTIPLE_CHOICE_MAPPING_NAMES
+                    and model.__class__.__name__ == MODEL_FOR_MULTIPLE_CHOICE_MAPPING_NAMES[model.config.model_type]
+                ):
+                    batch_size = inputs[next(iter(inputs))].shape[0]
+                    num_choices = inputs[next(iter(inputs))].shape[1]
+                    sequence_length = inputs[next(iter(inputs))].shape[2]
```

Comment:
* No need to create the same iterable repeatedly
* This seems like a pretty dangerous assumption that the first key always maps to the value we want here. Couldn't we just key on "input_ids" directly?

```suggestion
                    batch_size, num_choices, sequence_length, *_ = inputs["input_ids"].shape
```